src/gen/conditions.rs

```diff
@@ -280,7 +280,7 @@ pub fn parse_args(
     }
 }
 
-#[derive(Debug)]
+#[derive(Debug, Ord, PartialOrd)]
```

Comment:
this is actually problematic. the `NodePtr` is not deterministic, but is essentially a pointer to the hint, not the hint itself

Reply:
I'll have to figure something out